# Monte-Carlo Ternary Fission Simulator

This cell imports the necessary packages for the code
as well as a library which contains the code itself

In [1]:
%pylab inline
import numpy as np
import matplotlib.pyplot as plt
from mcternary import *

Populating the interactive namespace from numpy and matplotlib


The following is the main function of the program. It
takes the mass number of the desired nucleus and 
number of events. It also takes the names of the arrays
which contain the variables used throughout the code.
Using these inputs, the function calls all of the helper
functions in the library, and executes them in the correct
sequence. Subsequently, the function returns the initial
and final values of the coordinates, velocities and
momenta of each of the fragments.

In [2]:
runMCTernary(0, 4, alphapars, Lpars, Hpars, mandZandA, paxisandpP, dD, DEpars, hatpars, energies)

8.92773891679
21.2278633719
23.2241881485
7.92773891679
20.2278633719
22.2241881485
6.92773891679
19.2278633719
21.2241881485
5.92773891679
18.2278633719
20.2241881485
4.92773891679
17.2278633719
19.2241881485
3.92773891679
16.2278633719
18.2241881485
2.92773891679
15.2278633719
17.2241881485
1.92773891679
14.2278633719
16.2241881485
0.927738916793
13.2278633719
15.2241881485
-0.0722610832073
12.2278633719
14.2241881485
Results
8.92773891679
20.4280036312
22.6813772484
7.92773891679
19.4280036312
21.6813772484
6.92773891679
18.4280036312
20.6813772484
5.92773891679
17.4280036312
19.6813772484
4.92773891679
16.4280036312
18.6813772484
3.92773891679
15.4280036312
17.6813772484
2.92773891679
14.4280036312
16.6813772484
1.92773891679
13.4280036312
15.6813772484
0.927738916793
12.4280036312
14.6813772484
-0.0722610832073
11.4280036312
13.6813772484
Results
8.92773891679
21.3393830834
22.9168217424
7.92773891679
20.3393830834
21.9168217424
6.92773891679
19.3393830834
20.9168217424
5.92773891

This program follows the methods presented by Radi et al. in Monte Carlo Studies of Alpha-accompanied
Fission. It does so by taking the mass number, A, of the desired nucleus along with the number of events, 
and going through the following steps:

1. Performing a Monte-Carlo simulation for the mass number and the charge of the light fragment, AL, ZL.
    Values for mu and sigma are obtained from (reference TBD)
2. Using the semi-empirical mass formula to find the mass of the light fragment, mL.
3. Given a fixed A, and the obtained values for AL and ZL, the we get the mass number and charge of the 
    heavy fragment, assuming Aalapha and Zalpha to be 4 and 2, respectively.
4. Using the semi-empirical mass formula to obtain mH. (and alpha?)
5. (energies at scission by MC)
6. Using equation 21 from the paper to find values for d0 and D0 which will later be used to find the separation
    distance between L and the alpha plane, and the distance between L and H, d and D.
7. Running a MC simulation of d and D using d0 and D0 as mu, sigma_D=1 and sigma_d=sigma_xp=0.93.
8. Using equation 29 to find the coordinates of the prime axes, and setting the distance between alpha and the
    x-axis, Rc, equal to y'.
9. Using equaion 30 to find the momentum about the prime axes.
10. Calculating the initial velocities of L and H, VL0 and VH0 from the conservation of momentum.
11. Using equations 11-13 to calculate the constants, R0, V0, and T0. We obtain valuses for RL and RH from
    (reference TBD).
12. Calculating the initial coordinates and velocity components for the three fragments using equations from
    table I.
13. Using equations 7-10 to get the dimentionless variables required for the differential equations.
14. We get the total momenta of the three fragments and the distance between alpha and both, L and H. These
    will be used later to assesss the results of solving the equations of motion.
15. A function is written that calls all of the previous functions. This function initializes all the paramaters
    before they are passed into the numerical solver.
16. Another function is written that solves the differentail equations once the initial parameters are passed to it.
17. We write a third function that writes the results into file.
18. All of these functions are called within the main function which executes them in the correct sequence.
    While assessing the results coming out of the numerical solver. Where if the position of alpha passes
    within 0.8 of the radius of either of the other fragments, the program discards that event, and goes back
    to initialize new parameters. Additionally, the program checks the change in the momentum of the alpha
    particle, and to stop or continue calculating accordingly. Finally, this main function writes into file the
    initial and final values of the coordinates, velocities and momenta of the three fragments.